### PR TITLE
Add function plotsFromExcel

### PR DESCRIPTION
**Pull request by hannaei**
and add field `plotsFile` to `ProjectConfiguration` class

### CODE REVIEW

File: R/utilities-figures.R [hunk 1]
```diff
@@ -223,3 +223,129 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+          groups = simulated[j,]$group,
+          names = simulated[j,]$label)
+      }
+    }
+
+    # add observed data
+    observed <- dplyr::filter(dfDataCombined, DataCombinedName == name, dataType == "observed")
+    if (nrow(observed) > 0) {
+      dataSets <- observedData[observed$dataSet]
+      dataCombinedList[[name]]$addDataSets(dataSets, groups = observed$group)
```
Comment (PavelBal) [inline R/utilities-figures.R:280]: We will need tests for:

- no groups specified for observed data
- Same group for all obs data
- Different groups
- Some obs data with groups, some without.

File: R/utilities-figures.R [hunk 2]
```diff
@@ -223,3 +223,129 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+    }
+  }
+
+  # read sheet "plotConfiguration"
+  dfPlotConfigurations <- readExcel(projectConfiguration$plotsFile, sheet = "plotConfiguration")
+  # remove dataCombineds that have not been generated due to missing data
```
Comment (PavelBal) [inline R/utilities-figures.R:286]: `dataCombined**s**`? :D

File: R/utilities-figures.R [hunk 3]
```diff
@@ -223,3 +223,129 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+      warning(messages$warningInvalidPlotID(invalidPlotIDs, row$title))
+    }
+    plotGrid(plotGridConfiguration)
+  })
+  names(multiPanelPlots) <- dfPlotGrids$title
+
+  return(multiPanelPlots)
+}
+
+.plotConfigurationFromRow <- function(...) {
```
Comment (PavelBal) [inline R/utilities-figures.R:335]: Add roxygen documentation with `@keywords internal`

**Review by PavelBal — CHANGES_REQUESTED** [review]

**Comment by PavelBal** [comment]
Additionally:

- Update `NEWS`.
- Add a simple working example to `Plots.xlsx` and to the default test scenario.
- Run `styler`
- Add a sheet for `plotTypes` (pre-selection)

File: R/utilities-figures.R [hunk 4]
```diff
@@ -223,3 +223,129 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+    # add data to DataCombined object
+    # add simulated data
+    simulated <- dplyr::filter(dfDataCombined, DataCombinedName == name, dataType == "simulated")
+    if (nrow(simulated) > 0) {
+      for (j in 1:nrow(simulated)) {
+        dataCombinedList[[name]]$addSimulationResults(
+          simulationResults = simulatedScenarios[[simulated[j,]$scenario]]$results,
+          quantitiesOrPaths = simulated[j,]$path,
+          groups = simulated[j,]$group,
+          names = simulated[j,]$label)
```
[the remark this answers is not in the document]
Comment (PavelBal) [reply R/utilities-figures.R:272]: Update: `label` should be mandatory. Introduce a check and throw an error, if no label has been specified.

To make it obvious, move `label` column to column C.

File: R/utilities-figures.R [hunk 5]
```diff
@@ -223,3 +223,160 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+  missingScenarios <- setdiff(setdiff(dfDataCombined$scenario, names(simulatedScenarios)), NA)
+  if (length(missingScenarios) != 0) {
+    if (stopIfNotFound) {error(messages$stopInvalidScenarioName(missingScenarios))}
+    warning(messages$warningInvalidScenarioName(missingScenarios, projectConfiguration$plotsFile))
+    dfDataCombined <- dfDataCombined[!(dfDataCombined$scenario %in% missingScenarios), ]
+  }
+  # data set name not present in observedData
+  missingDataSets <- setdiff(setdiff(dfDataCombined$dataSet, names(observedData)), NA)
+  if (length(missingDataSets) != 0) {
+    if (stopIfNotFound) {error(messages$stopInvalidDataSetName(missingDataSets))}
```
Comment (PavelBal) [inline R/utilities-figures.R:257]: `error`, had too much Python coding? :D It should be `stop` ;) (or does it really work with `error`?)

File: R/utilities-figures.R [hunk 6]
```diff
@@ -223,3 +223,160 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
...
+#' simulated results or observed data are not found. If FALSE a warning is printed.
+#'
+#'
+#' @return A list of `ggplot` objects
+#'
+#' @export
+createPlotsFromExcel <- function(simulatedScenarios, observedData, projectConfiguration, stopIfNotFound = TRUE) {
+  validateIsOfType(observedData, "DataSet", nullAllowed = TRUE)
+  validateIsOfType(projectConfiguration, "ProjectConfiguration")
+  dfDataCombined <- readExcel(projectConfiguration$plotsFile, sheet = "DataCombined")
```
Comment (PavelBal) [inline R/utilities-figures.R:244]: It should be `dfDataCombined <- readExcel(file.path(projectConfiguration$paramsFolder, projectConfiguration$plotsFile), sheet = "DataCombined")`

File: R/utilities-figures.R (hunk 5 above)
Comment (hannaei) [reply R/utilities-figures.R:257]: Of course, produces an error as intended ;)

**Pull request merged**